sklearn/datasets/svmlight_format.py

```diff
@@ -179,7 +179,6 @@ def _open_and_load(f, dtype, multilabel, zero_based, query_id,
         actual_dtype, data, ind, indptr, labels, query = \
             _load_svmlight_file(f, dtype, multilabel, zero_based, query_id,
                                 offset, length)
-    # XXX remove closing when Python 2.7+/3.1+ required
```

Comment:
I don't get this

Reply:
@amueller See my comment in https://github.com/scikit-learn/scikit-learn/pull/12253#discussion_r222093897

Reply:
hm I feel like ``_gen_open`` should be a context manager then?

Reply:
Spend a few minutes trying to make it a context manager: that would require handling closing of the opened handled manually, and in this case it's not that straightforwards and there are 4 fh coming from 2 unrelated `if / elif` blocs with some code in between.  

In the end, I think `closing(_gen_open(f))` keeps the code simplest. It's part of the stdlib, so it's not much different to import `contextlib.contextmanager` or `contextlib.closing`.